arc-agents/src/main/kotlin/functions/LLMFunctionProvider.kt

```diff
@@ -31,8 +31,20 @@ class CompositeLLMFunctionProvider(
     private val functions: List<LLMFunction>,
 ) : LLMFunctionProvider {
 
+    /**
+     * Retrieves a list of LLMFunctions matching the given function name.
+     *
+     * @param functionName The name of the function to search for.
+     * @return List of LLMFunctions matching the function name.
+     * @throws NoSuchElementException if no matching LLMFunction is found.
+     */
+
     override fun provide(functionName: String): List<LLMFunction> {
-        return functions().filter { it.name == functionName }
+        val matchingFunctions = functions().filter { it.name == functionName }
+        if (matchingFunctions.isEmpty()) {
+            throw NoSuchElementException("No matching LLMFunction found for name: $functionName")
```

Comment:
Can we make this an explicit type, for example, FunctionNotFoundException?

Reply:
done